lib/childprocess.rb

```diff
@@ -168,6 +168,18 @@ def warn_once(msg)
       end
     end
 
+    # Workaround: detect the situation that an older Darwin Ruby is actually
+    # 64-bit, but is misreporting cpu as i686, which would imply 32-bit.
+    #
+    # @return [Boolean] `true` if:
+    #   (a) on Mac OS X
+    #   (b) on an older Ruby version
+    #   (c) actually running in 64-bit mode
+    def workaround_older_macosx_misreported_cpu?
+      os == :macosx && RUBY_VERSION < '2.4' && is_64_bit?
```

Comment:
This encapsulation works. (b) was never a factor of the OSX problem, so if we aren't checking fro Fixnums then we should be able to remove (b) entirely.